### PR TITLE
feat: [#188489128] non-profit question removal

### PR DESCRIPTION
**Pull request by adidner**
## Description

Removed non-profit question from onboarding as it's something that can be filled out in the post onboarding legal structure selection part

### Ticket

This pull request resolves [#188489128](https://www.pivotaltracker.com/story/show/188489128).

### Steps to Test

You should be able to, go through onboarding and see the question is gone

Got and select the non-profit legal structure and see that it still addes corresponding non-profit questions to roadmap

Trustees, charitable tax exempt, 

## Code author checklist

- [X] I have rebased this branch from the latest main branch
- [X] I have performed a self-review of my code
- [X] I have created and/or updated relevant documentation on the engineering documentation website
- [X] I have not used any relative imports
- [X] I have pruned any instances of unused code
- [X] I have not added any markdown to labels, titles and button text in config
- [X] If I added/updated any values in `userData` (including `profileData`, `formationData` etc), then I added a new migration file
- [X] I have checked for and removed instances of unused config from CMS
- [X] If I added any new collections to the CMS config, then I updated the search tool and `cmsCollections.ts` (see CMS Additions in Engineering Reference/FAQ on the engineering documentation site)
- [X] I have updated relevant `.env` values in both `.env-template` and in Bitwarden

### CODE REVIEW

File: api/src/db/migrations/v145_remove_nonprofit_onboarding.ts [hunk 1]
```diff
@@ -0,0 +1,805 @@
...
+    version: 145,
+  } as v145UserData;
+};
+
+export const migrate_v144Business_to_v145Business = (business: v144Business): v145Business => {
+  return {
+    ...business,
+    profileData: {
+      ...business.profileData,
+      isNonprofitOnboardingRadio: undefined,
```
Comment (adidner) [inline api/src/db/migrations/v145_remove_nonprofit_onboarding.ts:21]: I considered writing a test here and even started but found the typing throwing up at us. Plus it felt more like adding a field in the sense that it feels redundant to check a value that literally doesn't matter or do anything at the moment.

File: api/src/db/migrations/migrations.ts [hunk 2]
```diff
@@ -292,6 +293,7 @@ export const Migrations: MigrationFunction[] = [
   migrate_v141_to_v142,
   migrate_v142_to_v143,
   migrate_v143_to_v144,
+  migrate_v144_to_v145,
 ];
 
 export { generatev144UserData as CURRENT_GENERATOR } from "@db/migrations/v144_add_raffle_bingo_games";
```
Comment (SirSamuelJoseph) [inline api/src/db/migrations/migrations.ts:299]: Will have to update this as well, which I forgot to do!

**Review by SirSamuelJoseph — APPROVED** [review]
LGTM, just have to update the migration number and generator

File: api/src/db/migrations/v145_remove_nonprofit_onboarding.ts [hunk 3]
```diff
@@ -0,0 +1,805 @@
+import { v144Business, v144UserData } from "@db/migrations/v144_add_raffle_bingo_games";
+import { randomInt } from "@shared/intHelpers";
+
+export const migrate_v144_to_v145 = (v144Data: v144UserData): v145UserData => {
+  return {
+    ...v144Data,
+    businesses: Object.fromEntries(
+      Object.values(v144Data.businesses)
+        .map((business: v144Business) => migrate_v144Business_to_v145Business(business))
+        .map((currBusiness: v145Business) => [currBusiness.id, currBusiness])
```
Comment (cbrunefearless) [inline api/src/db/migrations/v145_remove_nonprofit_onboarding.ts:10]: you will have to update it to v147 now.. mine just went in with 145 from a previous commit and mine is 146

File: shared/src/test/factories.ts [hunk 4]
```diff
@@ -291,7 +290,6 @@ export const generateProfileData = (
     nexusDbaName: "",
     needsNexusDbaName: false,
     operatingPhase: OperatingPhaseId.NEEDS_TO_FORM,
-    isNonprofitOnboardingRadio: legalStructure === "nonprofit",
     nonEssentialRadioAnswers: {},
     elevatorOwningBusiness: undefined,
```
Comment (cbrunefearless) [inline shared/src/test/factories.ts:294]: nice

**Review by cbrunefearless — APPROVED** [review]
lgtm

**Pull request merged**